src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1155,9 +1169,9 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
                 {
                     if (fep.FormalParams[i].rank == Constants.kArbitraryRank)
                         noArbitraries++;
-
-                    numberOfArbitraryRanks.Add(noArbitraries);
```

Comment:
this was actually the source of the crash

Reply:
just curious, how was this the source of the crash?

Reply:
Looking at the code, it seemed to me that the intention was to record the number of arbitrary ranks ( 😄 ) for each function endpoint, not the weird logic that was there before.
And it crashed before because the function endpoints did not have any parameters at all, so numberOfArbitraryRanks was left empty => which in turn caused out of bounds index later on.